c7n/filters/core.py

```diff
@@ -83,6 +83,32 @@ def intersect(x, y):
     return bool(set(x).intersection(y))
 
 
+def cidr_overlap(x, y):
+
+    if isinstance(y, list):
+        for cidr in y:
+            if cidr_overlap(x, cidr):
+                return True
+        return False
+
+    if not isinstance(x, ipaddress._BaseNetwork) or not isinstance(y, ipaddress._BaseNetwork):
+        raise PolicyExecutionError(
```

Comment:
given there is implicit conversion and an additional validation check, and we don't really want runtime errors, what's the value here vs just letting it fall through to the attribute error below.

Reply:
thanks, yeah i forgot to remove that after setting value type by default